File: hw/palacios/u3/HW19AbstractFactory/HW19AbstractFactory-JS/controller/LinuxFactory.js
const GUIFactory = require('./GUIFactory');
const LinuxButton = require('../view/LinuxButton');
const LinuxMenu = require('../model/LinuxMenu');

class LinuxFactory extends GUIFactory {
    createButton() {
        return new LinuxButton();
    }

    createMenu() {
        return new LinuxMenu();
    }
}

module.exports = LinuxFactory;

File: hw/palacios/u3/HW19AbstractFactory/HW19AbstractFactory/AbstractFactory/src/main/java/ec/edu/espe/abstractfactory/controller/LinuxFactory.java
package ec.edu.espe.abstractfactory.controller;

import ec.edu.espe.abstractfactory.model.Button;
import ec.edu.espe.abstractfactory.model.LinuxMenu;
import ec.edu.espe.abstractfactory.model.Menu;
import ec.edu.espe.abstractfactory.view.LinuxButton;

/**
 *
 * @author Daniel Palacios
 */
public class LinuxFactory extends GUIFactory{

    @Override
    public Button createButton() {
        return (new LinuxButton());
    }

    @Override
    public Menu createMenu() {
        return (new LinuxMenu());
    }
}
